src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -11,89 +11,42 @@ namespace Bicep.Core.Emit
 {
     public class ResourceDependencyVisitor : SyntaxVisitor
     {
+        // shared data structure across all ResourceDependencyVisitors; our only "state" during traversals
+        // Instead of traversing through variables repeatedly, we pre-calculate variable-resource dependencies before we build the graph
+        private static Dictionary<VariableSymbol, ImmutableHashSet<DeclaredSymbol>> variableDependencies = new Dictionary<VariableSymbol, ImmutableHashSet<DeclaredSymbol>>();
```

Comment:
I think we should avoid using a static here as it'll outlive the lifetime of the visitor itself. The optimization itself makes sense (avoid visiting things we've already visited), but I think we should find a way to do it that doesn't involve this being static.

In the language server especially, this dictionary is going to just keep growing, as every time you type, the whole syntax tree & symbols get recomputed.